src/App.tsx

```diff
@@ -1,62 +1,135 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
 
-import classNames from 'classnames';
-import { PostsList } from './components/PostsList';
-import { PostDetails } from './components/PostDetails';
-import { UserSelector } from './components/UserSelector';
+import { UserSelector } from './components/UserSelector/UserSelector';
+import { client } from './utils/fetchClient';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { PostsList } from './components/PostsList/PostsList';
+import { Notification } from './types/Notification';
+import {
+  NotificationMessage,
+} from './components/NotificationMessage/NotificationMessage';
 import { Loader } from './components/Loader';
+import { Sidebar } from './components/Sidebar/Sidebar';
 
 export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(false);
+
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User>();
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+
+  const [error, setError] = useState<Notification>(Notification.NoError);
+  const [noPosts, setNoPosts] = useState(false);
+  const [loadingError, setLoadingError] = useState(false);
```

Comment:
Do you really need a state for it? You can just use "error" instead and check if the value is not "NoError"